src/Compiler/Checking/CheckDeclarations.fs

```diff
@@ -3945,7 +3963,21 @@ module TcDeclarations =
              | SynMemberDefn.NestedType (range=m) :: _ -> errorR(Error(FSComp.SR.tcTypesCannotContainNestedTypes(), m))
              | _ -> ()
         | ds ->
-            // Classic class construction 
+            // Check for duplicated parameters
+            for slot in ds do
+                if isAbstractSlot slot then
+                    match slot with
+                    | SynMemberDefn.AbstractSlot (x, _, m) ->
+                        let argNames =
```

Comment:
Please factor out the common code between the two

Reply:
Created a new function with the common code to reduce duplication :)

Reply:
Done . created a new function with the common code